testing/jormungandr-testing-utils/src/testing/node/grpc/client.rs

```diff
@@ -303,3 +308,159 @@ impl JormungandrClient {
         })
     }
 }
+
+pub struct JormungandrWatchClient {
+    addr: SocketAddr,
+    inner_client: WatchClient<Channel>,
+    rt: Arc<Runtime>,
+}
+
+impl Clone for JormungandrWatchClient {
+    fn clone(&self) -> Self {
+        JormungandrWatchClient::new(self.addr)
+    }
+}
+
+impl fmt::Debug for JormungandrWatchClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("JormungandrWatchClient")
+            .field("host", &self.addr)
+            .finish()
+    }
+}
+
+pub type WatchTipNotifier = Arc<(Mutex<Result<LibHeader, tonic::Status>>, Condvar)>;
+
+impl JormungandrWatchClient {
+    pub fn from_address(address: &str) -> Result<Self, MockClientError> {
+        let addr = address
+            .parse()
+            .map_err(|_| MockClientError::InvalidAddressFormat(address.to_owned()))?;
+        Ok(Self::new(addr))
+    }
+
+    pub fn new(addr: SocketAddr) -> Self {
+        let rt = Arc::new(Builder::new_current_thread().enable_all().build().unwrap());
+        let inner_client = rt.block_on(async {
+            WatchClient::new(
+                tonic::transport::Endpoint::from_shared(format!("http://{}", addr))
+                    .unwrap()
+                    .connect_lazy(),
+            )
+        });
+
+        Self {
+            addr,
+            inner_client,
+            rt,
+        }
+    }
+
+    fn client(&self) -> WatchClient<Channel> {
+        self.inner_client.clone()
+    }
+
+    pub fn tip_subscription(&self) -> WatchTipNotifier {
```

Comment:
Old plain mutex are used instead of async stuff because the testing framework is not very async friendly?

Reply:
I would say is directly not async, all the other grpc functions use `block_on` internally to make the api sync, so I tried to keep it that way.

I wonder if we can use `#[tokio::test]` in the test functions instead and just use streams here.

Reply:
> I wonder if we can use #[tokio::test] in the test functions instead and just use streams here.

And I think the answer is probably not without doing a big refactor that doesn't belong in this PR. But I think we should remove the runtime from both clients, and just leave it to the callers to deal with the executor. For the client tests we can use `#[tokio::test]`, and the rest of the code can just wrap things in `futures::executor::block_on`.

Or we could rewrite things in terms of `spawn`, instead of `block_on`, maybe with `tokio::runtime::Handle::try_current`.